pkg/build/graph/analysis/bc.go

```diff
@@ -185,3 +284,17 @@ func latestBuild(g osgraph.Graph, bc graph.Node) *buildgraph.BuildNode {
 	sort.Sort(sort.Reverse(buildapi.BuildPtrSliceByCreationTimestamp(builds)))
 	return buildNameToNode[builds[0].Name]
 }
+
+// addInputMarker updates the Marker array ptr input parameter with a new marker constructed from the remaining parameters
+func addInputMarker(markers *[]osgraph.Marker, bcNode graph.Node, istNode graph.Node, imageStreamNode *imagegraph.ImageStreamNode, key string, message string, suggestion osgraph.Suggestion) {
```

Comment:
I'm guessing the names are bad and you allow an ISI in the ist location.  This only adds a warning.  Tighten the signature if you want this helper.

`addBuildConfigWarning(buildgraph.bcNode, imaggraph.ImageStreamNode, otherrelatedNodes []graph.Node, etc)`